src/saltext/vmware/states/esxi.py

```diff
@@ -877,12 +878,13 @@ def lockdown_mode(
 
 
 def advanced_config(
-    name,
-    value,
+    name=None,
+    value=None,
     datacenter_name=None,
     cluster_name=None,
     host_name=None,
     service_instance=None,
+    profile=None,
 ):
```

Comment:
**suggestion** let's go ahead and remove advanced_config in favor of `advanced_configs`. Just add `332.removed` to the changelog, probably just `Removed advanced_config in favor of advanced_configs.` And we can pretty easily link to documentation from this PR for anyone who might actually need it :+1:

Reply:
Not strictly necessary, since we're removing this function, but: `name` will always be provided (in the example in this docstring it will be "Remove User"), so it shouldn't be given a default of `None`. Also, note that in the docstring `name` and `value` are marked `(required)`, so they shouldn't have a default value unless within the body of the function a check happens.

Reply:
As advanced_config is removed, this changes are valid in the new advanced_configs